test/crdsvalidation/konnect.konghq.com/konnectextension_test.go

```diff
@@ -127,22 +127,6 @@ func TestKonnectExtension(t *testing.T) {
 				},
 				ExpectedErrorMessage: lo.ToPtr("secretRef must not be set when provisioning is set to Automatic."),
 			},
-			{
-				Name: "kic controlplane",
-				TestObject: &konnectv1alpha2.KonnectExtension{
-					ObjectMeta: common.CommonObjectMeta,
-					Spec: konnectv1alpha2.KonnectExtensionSpec{
-						Konnect: konnectv1alpha2.KonnectExtensionKonnectSpec{
-							ControlPlane: konnectv1alpha2.KonnectExtensionControlPlane{
-								Ref: commonv1alpha1.ControlPlaneRef{
-									Type: configurationv1alpha1.ControlPlaneRefKIC,
-								},
-							},
-						},
-					},
-				},
-				ExpectedErrorMessage: lo.ToPtr("kic type not supported as controlPlaneRef."),
-			},
```

Comment:
Wouldn't it make sense to actually leave this in (perhaps with a comment), just to ensure that we don't allow this? It's just a matter of adjusting the error, right? Based on the validation in `KonnectExtensionControlPlaneRef` AFACT

Reply:
Alright, that makes sense. PTAL